Source/AssetRipper.Processing/AnimationClips/AnimationClipConverter.cs

```diff
@@ -524,41 +529,62 @@ private void AddPPtrKeyframe(in CurveData curveData, IAnimationClipBindingConsta
 				m_pptrs.Add(curveData, curve);
 			}
 
-			IPPtr_Object value = bindings.PptrCurveMapping[index];
+			IPPtr_Object value = m_clip.ClipBindingConstant_C74!.PptrCurveMapping[index];
 			IPPtrKeyframe key = curve.Curve.AddNew();
 			key.Time = time;
 			key.Value.CopyValues(value, new PPtrConverter(m_clip));
 		}
 
-		private static void GetPreviousFrame(IReadOnlyList<StreamedFrame> streamFrames, int curveID, int currentFrame, out int frameIndex, out int curveIndex)
+		private IGenericBinding GetBinding(int index)
 		{
-			for (frameIndex = currentFrame - 1; frameIndex >= 0; frameIndex--)
+			if (m_bindingsCache.TryGetValue(index, out IGenericBinding? binding))
 			{
-				StreamedFrame frame = streamFrames[frameIndex];
-				for (curveIndex = 0; curveIndex < frame.Curves.Length; curveIndex++)
+				return binding;
+			}
+			int curves = 0;
+			AccessListBase<IGenericBinding> bindings = m_clip.ClipBindingConstant_C74!.GenericBindings;
```

Comment:
I don't like the implicit assumption that `ClipBindingContant_C74` is not null. I know the current code ensures that's the always true, but I'd like something more explicit.
```cs
IAnimationClipBindingConstant ClipBindingConstant => m_clip.ClipBindingConstant_C74!;//This class only supports 5.3 and newer.
```

Reply:
implemented